kinto/core/resource/__init__.py

```diff
@@ -1184,18 +1184,29 @@ def process_record(self, new, old=None):
         """
         new = super().process_record(new, old)
 
-        # patch is specified as a list of of operations (RFC 6902)
+        # patch is specified as a list of of operations (RFC i6902)
+
+        payload = self.request.validated['body']
+
         if self._is_json_patch:
-            changes = self.request.validated['body']
-            permissions = apply_json_patch(old, changes)['permissions']
+            permissions = apply_json_patch(old, payload)['permissions']
+
+        elif self._is_merge_patch:
+            existing = old or {}
+            permissions = existing.get('__permissions__', {})
+            recursive_update_dict(permissions,
+                                  payload.get('permissions', {}),
+                                  ignores=(None,))
+
         else:
-            permissions = self.request.validated['body'].get('permissions', {})
+            permissions = {k: v for k, v in payload.get('permissions', {}).items()
+                           if v is not None}
 
         annotated = {**new}
 
         if permissions:
```

Comment:
This is outside the scope of this PR, but I guess this means we can never set the permissions to the empty object.

Reply:
Actually, this change is meaningful to the fix. Without it a null value on a regular patch would remove permissions as well.

Reply:
Yes, but IMHO passing an empty object should be different from not passing anything.